sdk/appconfiguration/azure-data-appconfiguration/src/samples/java/com/azure/data/appconfiguration/ConfigurationClientJavaDocCodeSnippets.java

```diff
@@ -275,45 +276,59 @@ public void deleteSetting() {
+        SettingSelector settingSelector = new SettingSelector().keys("prodDBConnection");
+        configurationClient.listSettings(settingSelector).forEach(setting -> {
             System.out.printf("Key: %s, Value: %s", setting.key(), setting.value());
-        }
-        // END: com.azure.data.applicationconfig.configurationclient.listSettings#SettingSelector
+        });
+        // END: com.azure.data.applicationconfig.configurationclient.listSettings#settingSelector
+    }
 
-        /**
-         * Generates code sample for using {@link ConfigurationClient#listSettings(SettingSelector, Context)}
-         */
-        // BEGIN: com.azure.data.applicationconfig.configurationclient.listSettings#SettingSelector-Context
-        for (ConfigurationSetting setting : configurationClient.listSettings(new SettingSelector()
-            .keys("prodDBConnection"), new Context(key1, value1))) {
+    /**
+     * Generates code sample for using {@link ConfigurationClient#listSettings(SettingSelector, Context)}
+     */
+    public void listSettingsContext() {
+        ConfigurationClient configurationClient = createSyncConfigurationClient();
+        // BEGIN: com.azure.data.applicationconfig.configurationclient.listSettings#settingSelector-context
+        SettingSelector settingSelector = new SettingSelector().keys("prodDBConnection");
+        Context ctx = new Context(key2, value2);
+        configurationClient.listSettings(settingSelector, ctx).forEach(setting -> {
             System.out.printf("Key: %s, Value: %s", setting.key(), setting.value());
-        }
-        // END: com.azure.data.applicationconfig.configurationclient.listSettings#SettingSelector-Context
+        });
+        // END: com.azure.data.applicationconfig.configurationclient.listSettings#settingSelector-context
     }
 
     /**
      * Generates code sample for using {@link ConfigurationClient#listSettingRevisions(SettingSelector)}
      */
     public void listSettingRevisions() {
-        ConfigurationClient configurationClient = createSyncConfigurationClient();
-        // BEGIN: com.azure.data.applicationconfig.configurationclient.listSettingRevisions#SettingSelector
-        for (ConfigurationSetting revision : configurationClient.listSettingRevisions(new SettingSelector()
-            .keys("prodDBConnection"))) {
-            System.out.printf("Key: %s, Value: %s", revision.key(), revision.value());
-        }
-        // END: com.azure.data.applicationconfig.configurationclient.listSettingRevisions#SettingSelector
+        ConfigurationClient client = createSyncConfigurationClient();
+        // BEGIN: com.azure.data.applicationconfig.configurationclient.listSettingRevisions#settingSelector
+        SettingSelector settingSelector = new SettingSelector().keys("prodDBConnection");
+        client.listSettingRevisions(settingSelector).streamByPage().forEach(response -> {
+            if (response.statusCode() == HttpURLConnection.HTTP_OK) {
```

Comment:
What actually happens when the underlying `PagedFlux` has an error response? I would think that an exception is thrown instead of having a return like this.

Reply:
I would agree.